sdk/core/azure-core/src/http/curl/curl.cpp

```diff
@@ -323,6 +323,13 @@ void CurlSession::ParseChunkSize()
         // get chunk size. Chunk size comes in Hex value
         this->m_chunkSize = static_cast<int64_t>(std::stoull(strChunkSize, nullptr, 16));
 
+        if (this->m_chunkSize == 0)
```

Comment:
Unit tests?

Reply:
it requires mock or finding some server to return a zero chunk data.
I tested it based on the issue reported with a premium storage account.
I will create a new issue to do not forget about adding unit tests for this once we have a way to mock network responses